### PR TITLE
media-libs/gegl: workaround to fix build on hardened gentoo systems

### PR DESCRIPTION
**Pull request by band-a-prend**
On gentoo hardened fatal error: opencl/* no such file or directory
takes place due to opencl/meson.build doesn't produce appropriate
headers from *.cl files.

Closes: https://bugs.gentoo.org/739816

### CODE REVIEW

File: media-libs/gegl/gegl-0.4.26-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+		-i tests/compositions/meson.build || die
+
+	# fix skipping mipmap tests due to executable not found
+	for item in "invert-crop.sh" "invert.sh" "rotate-crop.sh" "rotate.sh" "unsharp-crop.sh" "unsharp.sh"; do
+		sed -i "s:/bin/gegl:/bin/gegl-0.4:g" "${S}/tests/mipmap/${item}" || die
+		sed -i "s:/tools/gegl-imgcmp:/tools/gegl-imgcmp-0.4:g" "${S}/tests/mipmap/${item}" || die
+	done
+
+	# fix 'build'headers from *.cl on gentoo-hardened, bug 739816
+	pushd "${S}/opencl/"
```
Comment (juippis) [inline media-libs/gegl/gegl-0.4.26-r1.ebuild:110]: `|| die`

File: media-libs/gegl/gegl-0.4.26-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	for file in *.cl; do
+		if [ -f "$file" ]; then
+			python cltostring.py $file
```
Comment (juippis) [inline media-libs/gegl/gegl-0.4.26-r1.ebuild:113]: `${EPYTHON}` instead of `python` here?
`${file}`.

File: media-libs/gegl/gegl-0.4.26-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		fi
+	done
+	popd
```
Comment (juippis) [inline media-libs/gegl/gegl-0.4.26-r1.ebuild:116]: `|| die`.

**Review by juippis — COMMENTED** [review]
Well I did notice more issues in the ebuild, but decided to only focus on what you're attempting to fix. No need to rewrite the whole ebuild due to this.

File: media-libs/gegl/gegl-0.4.26-r1.ebuild (hunk 2 above)
Comment (band-a-prend) [reply media-libs/gegl/gegl-0.4.26-r1.ebuild:113]: add `${PYTHON_DEPS}` to `BDEPEND`;  
remove `pkg_setup()` as now it doesn't depend on `USE="test"`;  
add `|| die` also after `${file}`

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**